planetarium/NineChronicles
# Add PatchTableSheet tool

### Description

- Add editor menu - Tools/Print TablePatch Action Hash
  - Easy tool to get `PatchTableSheet` action hashes for table patches.
  - It refered [tx patch-table](https://github.com/planetarium/lib9c/blob/development/.Lib9c.Tools/SubCommand/Tx.cs#L124) in Lib9c tools.  

### Related Links

[테이블 패치 자동화](https://app.asana.com/0/958521740385861/1203013129498112/f)

### How to test, Screenshot

**Please refer to the [Notion Docs](https://www.notion.so/planetarium/43755b43a89c46238a7885e62a969670#b97530bbf1114a28b227951c2f687df5) for details.**

1. Input TableSheet files into `TableSheetsForPatch` scriptable object. 
  (Assets/Resources/ScriptableObject/TableSheetsForPatch.asset)
  ![image](https://user-images.githubusercontent.com/63496908/191906802-c4c15885-e79a-470f-980d-32553a4096c2.png)

2. Run menu `Tools/Print TablePatch Action Hash` in Unity Editor.
  ![image](https://user-images.githubusercontent.com/63496908/191906897-2f84590f-7c7b-479f-a887-2473cfdc222b.png)

3. Check the results, consol log and text file.
  (Assets/Editor/TablePatch Action Hash.txt)
  ![image](https://user-images.githubusercontent.com/63496908/191906923-eaf1dad5-9400-4d58-a19a-8ddf2303d213.png)
